routes/articles.js

```diff
@@ -9,26 +9,45 @@ router.post('/', async (req,res) => {
         title : req.body.title,
         description :  req.body.description,
         author : req.body.author,
-        tags :  req.body.tags
+        tags :  req.body.tags,
+        time : req.body.time
     })
     try {
         const newArticle = await article.save()
         //201 cause its sucess
         res.status(201).json(newArticle)
-    } catch(arr) {
-        res.status(400).json( { message : err.message })
+    } catch(err) {
+        //client's mistake
+        res.status(401).json( { message : err.message })
     }
 })
 
 
 //get all articles
 router.get('/' , async (req,res) => {
     try {
-        //todo--> add more serch and filter options
-        const articles = await Article.find()
+        //todo--> add article limits
+        let searchOptions = {}
+        if( req.query.title ) {
+            searchOptions.title = new RegExp(req.query.title, 'i')
+        }
+        if( req.query.author ) {
+            searchOptions.author = new RegExp(req.query.author, 'i')
+        }
+
+        //we can have some default values of these as well
+        if( req.query.toDate && req.query.fromDate ) {
+            searchOptions.time = {
+                $gte: new Date(req.query.fromDate), 
+                $lte: new Date(req.query.toDate)
+            } 
+        }
+              
+        let sortOptions = {}
+        sortOptions[ req.query.sortBy || "time" ] = req.query.orderBy === 'asnd' ? 1 : -1
```

Comment:
asnd doesn't looks a good name
`asc` is the normal convention
lets go with `asc` and `desc` as in normal sql queries

Reply:
Sorry my bad, will do that.